test-frame-common/src/main/java/io/skodjob/testframe/clients/cmdClient/BaseCmdKubeClient.java

```diff
@@ -0,0 +1,400 @@
+            Exec.exec(namespacedCommand(CREATE, "namespace", name));
+        }
+        return (K) this;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public K deleteNamespace(String name) {
+        try (Context context = adminContext()) {
+            Exec.exec(null, namespacedCommand(DELETE, "namespace", name), 0, true, false);
+        }
+        return (K) this;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public K scaleByName(String kind, String name, int replicas) {
+        try (Context context = defaultContext()) {
+            Exec.exec(null, namespacedCommand("scale", kind, name, "--replicas", Integer.toString(replicas)));
+            return (K) this;
+        }
+    }
+
+    @Override
+    public ExecResult execInPod(String pod, String... command) {
+        List<String> cmd = namespacedCommand("exec", pod, "--");
+        cmd.addAll(asList(command));
+        return Exec.exec(cmd);
+    }
+
+    @Override
+    public ExecResult execInPodContainer(String pod, String container, String... command) {
+        return execInPodContainer(true, pod, container, command);
+    }
+
+    @Override
+    public ExecResult execInPodContainer(boolean logToOutput, String pod, String container, String... command) {
+        List<String> cmd = namespacedCommand("exec", pod, "-c", container, "--");
+        cmd.addAll(asList(command));
+        return Exec.exec(null, cmd, 0, logToOutput);
+    }
+
+    @Override
+    public ExecResult exec(String... command) {
+        return exec(true, command);
+    }
+
+    @Override
+    public ExecResult exec(boolean throwError, String... command) {
+        return exec(throwError, true, command);
+    }
+
+    @Override
+    public ExecResult exec(boolean throwError, boolean logToOutput, String... command) {
+        List<String> cmd = command(asList(command), false);
+        return Exec.exec(null, cmd, 0, logToOutput, throwError);
+    }
+
+    @Override
+    public ExecResult execInCurrentNamespace(String... commands) {
+        return Exec.exec(namespacedCommand(commands));
+    }
+
+    @Override
+    public ExecResult execInCurrentNamespace(boolean logToOutput, String... commands) {
+        return Exec.exec(null, namespacedCommand(commands), 0, logToOutput);
+    }
+
+    enum ExType {
+        BREAK,
+        CONTINUE,
+        THROW
+    }
+
+    @Override
+    public String toString() {
+        return cmd();
+    }
+
+    @Override
+    public List<String> list(String resourceType) {
+        return Arrays.stream(Exec.exec(namespacedCommand("get", resourceType, "-o", "jsonpath={range .items[*]}{.metadata.name} ")).out().trim().split(" +"))
+                .filter(s -> !s.trim().isEmpty()).collect(Collectors.toList());
+    }
+
+    @Override
+    public String getResourceAsJson(String resourceType, String resourceName) {
+        return Exec.exec(namespacedCommand("get", resourceType, resourceName, "-o", "json")).out();
```

Comment:
maybe we should create some constants or something for `get`, `exec`, `scale`, .etc?